### PR TITLE
feat(mojaloop/#2352): update and standardise helm charts to use networking v1

### PR DESCRIPTION
**Pull request by mdebarros**
feat(mojaloop/#2352): update and standardise helm charts to use networking v1 - https://github.com/mojaloop/project/issues/2352

# Notes

This PR is re-created from https://github.com/mojaloop/helm/pull/506 which includes some fixes based on tests on k8s v1.24

# Changes:

## Update the Chart.yaml's and standardise all charts to helm v3 (API v2)
    - move any dependencies in the requirements.yaml to Charts.yaml
    - update the apiVersion for helm in all the Charts.yaml to 2.0
    - add the common dependency to each chart that already has an ingress
    - remove all requirements.lock 
    - update maintainers in chart.yaml to include tomd@crosslaketech.com
    - bump chart versions 

## Update the ingress and networking API to v1 and standardise where possible 
   -  if the chart has an ingress template the chart  is not on the exclusion list (see below) => the ingress template is updated to be a clone of the bitnami example (a version of what already exists in the charts repo)
   - updates the values files for the new ingress settings
   - ensure the updated values files have the correct hostname for the ingress
   - checks for paths and extra paths correctly allowed for (i.e. all charts with extraPaths or extraHosts are on e
   - ensure the updated values files have the correct port number for the ingress
    - update config/default.json files for values.ingress.api.host or similar to use .Values.ingress.hostname
    - _helper.tpl's updated to use correct ingress APIs and remove redundant ingress logic and templates

## General Maintenance
    - Fixed lint issue
    - Removed forensicloggingsidecar, centralenduserregistry, kube-system from lint scripts since they are no longer supported
    - Updated Maintainers list for each chart

## Exclusion list of charts
- i.e. these charts are not yet standardised to the new ingress modeled on the bitnami example :
- Any existing ingress in these charts has been updated "in place" rather than replaced with the newer standardised bitnami modeled chart.
- So the ingress.yaml, config/*json, _helper.tpls , values.yaml files etc have been updated to use the latest ingress but they remain non-standard compared to the rest of the code-base.
- "finance-portal-settlement-management",
- "finance-portal",
- ~"thirdparty"~,
- ~"thirdparty/chart-tp-api-svc"~,
- ~"thirdparty/chart-consent-oracle"~,
- ~"thirdparty/chart-auth-svc"~,
- ~"mojaloop-simulator"~,
- "keycloak",
- "monitoring",
- "monitoring/promfana",
- "monitoring/elk",
- "ml-testing-toolkit/chart-keycloak",
- "ml-testing-toolkit/chart-backend",
- "ml-testing-toolkit/chart-frontend",
- "ml-testing-toolkit/chart-connection-manager-backend",
- "ml-testing-toolkit/chart-connection-manager-frontend"

### CODE REVIEW

**Comment by mdebarros** [comment]
```
--------------------FINAL REPORT--------------------

Test Suite:Thirdparty Tests
Environment:Development
┌───────────────────────────────────────────────────┐
│                      SUMMARY                      │
├───────────────────┬───────────────────────────────┤
│ Total assertions  │ 21                            │
├───────────────────┼───────────────────────────────┤
│ Passed assertions │ 21                            │
├───────────────────┼───────────────────────────────┤
│ Failed assertions │ 0                             │
├───────────────────┼───────────────────────────────┤
│ Total requests    │ 20                            │
├───────────────────┼───────────────────────────────┤
│ Total test cases  │ 8                             │
├───────────────────┼───────────────────────────────┤
│ Passed percentage │ 100.00%                       │
├───────────────────┼───────────────────────────────┤
│ Started time      │ Wed, 19 Oct 2022 10:48:06 GMT │
├───────────────────┼───────────────────────────────┤
│ Completed time    │ Wed, 19 Oct 2022 10:48:14 GMT │
├───────────────────┼───────────────────────────────┤
│ Runtime duration  │ 7155 ms                       │
└───────────────────┴───────────────────────────────┘
TTK-Assertion-Report-multi-2022-10-19T10:48:14.113Z.html was generated
Terminate with exit code 0
```

**Comment by mdebarros** [comment]
```
--------------------FINAL REPORT--------------------

Test Suite:GP Tests
Environment:Development
┌───────────────────────────────────────────────────┐
│                      SUMMARY                      │
├───────────────────┬───────────────────────────────┤
│ Total assertions  │ 2634                          │
├───────────────────┼───────────────────────────────┤
│ Passed assertions │ 2634                          │
├───────────────────┼───────────────────────────────┤
│ Failed assertions │ 0                             │
├───────────────────┼───────────────────────────────┤
│ Total requests    │ 598                           │
├───────────────────┼───────────────────────────────┤
│ Total test cases  │ 130                           │
├───────────────────┼───────────────────────────────┤
│ Passed percentage │ 100.00%                       │
├───────────────────┼───────────────────────────────┤
│ Started time      │ Wed, 19 Oct 2022 12:45:55 GMT │
├───────────────────┼───────────────────────────────┤
│ Completed time    │ Wed, 19 Oct 2022 12:49:10 GMT │
├───────────────────┼───────────────────────────────┤
│ Runtime duration  │ 195465 ms                     │
└───────────────────┴───────────────────────────────┘
TTK-Assertion-Report-multi-2022-10-19T12:49:10.955Z.html was generated
```

```
--------------------FINAL REPORT--------------------

Test Suite:Thirdparty Tests
Environment:Development
┌───────────────────────────────────────────────────┐
│                      SUMMARY                      │
├───────────────────┬───────────────────────────────┤
│ Total assertions  │ 21                            │
├───────────────────┼───────────────────────────────┤
│ Passed assertions │ 21                            │
├───────────────────┼───────────────────────────────┤
│ Failed assertions │ 0                             │
├───────────────────┼───────────────────────────────┤
│ Total requests    │ 20                            │
├───────────────────┼───────────────────────────────┤
│ Total test cases  │ 8                             │
├───────────────────┼───────────────────────────────┤
│ Passed percentage │ 100.00%                       │
├───────────────────┼───────────────────────────────┤
│ Started time      │ Wed, 19 Oct 2022 10:48:06 GMT │
├───────────────────┼───────────────────────────────┤
│ Completed time    │ Wed, 19 Oct 2022 10:48:14 GMT │
├───────────────────┼───────────────────────────────┤
│ Runtime duration  │ 7155 ms                       │
└───────────────────┴───────────────────────────────┘
TTK-Assertion-Report-multi-2022-10-19T10:48:14.113Z.html was generated
Terminate with exit code 0
```

```
--------------------FINAL REPORT--------------------

Test Suite:Bulk Tests
Environment:Development
┌───────────────────────────────────────────────────┐
│                      SUMMARY                      │
├───────────────────┬───────────────────────────────┤
│ Total assertions  │ 183                           │
├───────────────────┼───────────────────────────────┤
│ Passed assertions │ 183                           │
├───────────────────┼───────────────────────────────┤
│ Failed assertions │ 0                             │
├───────────────────┼───────────────────────────────┤
│ Total requests    │ 17                            │
├───────────────────┼───────────────────────────────┤
│ Total test cases  │ 4                             │
├───────────────────┼───────────────────────────────┤
│ Passed percentage │ 100.00%                       │
├───────────────────┼───────────────────────────────┤
│ Started time      │ Wed, 19 Oct 2022 10:46:44 GMT │
├───────────────────┼───────────────────────────────┤
│ Completed time    │ Wed, 19 Oct 2022 10:47:10 GMT │
├───────────────────┼───────────────────────────────┤
│ Runtime duration  │ 25705 ms                      │
└───────────────────┴───────────────────────────────┘
TTK-Assertion-Report-multi-2022-10-19T10:47:10.164Z.html was generated
```

**Comment by mdebarros** [comment]
Moja3 Environment

```
❯ kubectl version
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.8", GitCommit:"5575935422cc1cf5169dfc8847cb587aa47bac5a", GitTreeState:"clean", BuildDate:"2021-06-16T12:53:07Z", GoVersion:"go1.15.13", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.25) and server (1.20) exceeds the supported minor version skew of +/-1
```

```
--------------------FINAL REPORT--------------------

Test Suite:GP Tests
Environment:Development
┌───────────────────────────────────────────────────┐
│                      SUMMARY                      │
├───────────────────┬───────────────────────────────┤
│ Total assertions  │ 2634                          │
├───────────────────┼───────────────────────────────┤
│ Passed assertions │ 2634                          │
├───────────────────┼───────────────────────────────┤
│ Failed assertions │ 0                             │
├───────────────────┼───────────────────────────────┤
│ Total requests    │ 598                           │
├───────────────────┼───────────────────────────────┤
│ Total test cases  │ 130                           │
├───────────────────┼───────────────────────────────┤
│ Passed percentage │ 100.00%                       │
├───────────────────┼───────────────────────────────┤
│ Started time      │ Wed, 19 Oct 2022 13:38:25 GMT │
├───────────────────┼───────────────────────────────┤
│ Completed time    │ Wed, 19 Oct 2022 13:41:40 GMT │
├───────────────────┼───────────────────────────────┤
│ Runtime duration  │ 195281 ms                     │
└───────────────────┴───────────────────────────────┘
TTK-Assertion-Report-multi-2022-10-19T13:41:40.624Z.html was generated
Terminate with exit code 0
```

File: finance-portal-settlement-management/Chart.yaml [hunk 1]
```diff
@@ -1,5 +1,16 @@
-apiVersion: v1
+apiVersion: v2
 description: "Portal Settlement (deprecated)"
 name: finance-portal-settlement-management
-version: 11.0.3
+version: 11.1.3
```
Comment (elnyry-sam-k) [inline finance-portal-settlement-management/Chart.yaml:4]:
```suggestion
version: 11.1.0
```

File: finance-portal/Chart.yaml [hunk 2]
```diff
@@ -1,5 +1,16 @@
-apiVersion: v1
+apiVersion: v2
 description: "Finance portal service (deprecated)"
 name: finance-portal
-version: 11.0.3
+version: 11.1.3
```
Comment (elnyry-sam-k) [inline finance-portal/Chart.yaml:4]:
```suggestion
version: 11.1.0
```

File: ml-operator/Chart.yaml [hunk 3]
```diff
@@ -2,13 +2,21 @@ apiVersion: v2
 name: ml-operator
 description: ml-operator makes running and managing a Mojaloop deployment easier
 type: application
-version: 0.1.1
+version: 0.2.1
```
Comment (elnyry-sam-k) [inline ml-operator/Chart.yaml:5]:
```suggestion
version: 0.2.0
```

**Comment by elnyry-sam-k** [comment]
Thanks @mdebarros .. looking good overall..

Thanks for the results snapshot, looks great..

Just couple of minor comments / nits on version number bumps and maybe revisit the "test" CI step.. happy to approve

**Comment by mdebarros** [comment]
# Moja3 Environment with BULK & THIRDPARTY

Default Mojaloop Install with Bulk and Thirdparty enabled.

## Environment

```
❯ helm version
version.BuildInfo{Version:"v3.10.0", GitCommit:"ce66412a723e4d89555dc67217607c6579ffcb21", GitTreeState:"clean", GoVersion:"go1.19.1"}
❯ kubectl version
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.8", GitCommit:"5575935422cc1cf5169dfc8847cb587aa47bac5a", GitTreeState:"clean", BuildDate:"2021-06-16T12:53:07Z", GoVersion:"go1.15.13", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.25) and server (1.20) exceeds the supported minor version skew of +/-1
```

## Results

```

```

```
--------------------FINAL REPORT--------------------

Test Suite:Bulk Tests
Environment:Development
┌───────────────────────────────────────────────────┐
│                      SUMMARY                      │
├───────────────────┬───────────────────────────────┤
│ Total assertions  │ 183                           │
├───────────────────┼───────────────────────────────┤
│ Passed assertions │ 183                           │
├───────────────────┼───────────────────────────────┤
│ Failed assertions │ 0                             │
├───────────────────┼───────────────────────────────┤
│ Total requests    │ 17                            │
├───────────────────┼───────────────────────────────┤
│ Total test cases  │ 4                             │
├───────────────────┼───────────────────────────────┤
│ Passed percentage │ 100.00%                       │
├───────────────────┼───────────────────────────────┤
│ Started time      │ Wed, 19 Oct 2022 14:14:09 GMT │
├───────────────────┼───────────────────────────────┤
│ Completed time    │ Wed, 19 Oct 2022 14:14:35 GMT │
├───────────────────┼───────────────────────────────┤
│ Runtime duration  │ 25571 ms                      │
└───────────────────┴───────────────────────────────┘
TTK-Assertion-Report-multi-2022-10-19T14:14:35.269Z.html was generated
```


```
--------------------FINAL REPORT--------------------

Test Suite:Thirdparty Tests
Environment:Development
┌───────────────────────────────────────────────────┐
│                      SUMMARY                      │
├───────────────────┬───────────────────────────────┤
│ Total assertions  │ 21                            │
├───────────────────┼───────────────────────────────┤
│ Passed assertions │ 21                            │
├───────────────────┼───────────────────────────────┤
│ Failed assertions │ 0                             │
├───────────────────┼───────────────────────────────┤
│ Total requests    │ 20                            │
├───────────────────┼───────────────────────────────┤
│ Total test cases  │ 8                             │
├───────────────────┼───────────────────────────────┤
│ Passed percentage │ 100.00%                       │
├───────────────────┼───────────────────────────────┤
│ Started time      │ Wed, 19 Oct 2022 14:15:40 GMT │
├───────────────────┼───────────────────────────────┤
│ Completed time    │ Wed, 19 Oct 2022 14:15:49 GMT │
├───────────────────┼───────────────────────────────┤
│ Runtime duration  │ 8131 ms                       │
└───────────────────┴───────────────────────────────┘
TTK-Assertion-Report-multi-2022-10-19T14:15:49.062Z.html was generated
```

**Review by elnyry-sam-k — DISMISSED** [review]

**Pull request merged**